### ISSUE TITLE
Make all task titles/descriptions read-only to all but the author

### ISSUE THREAD

**Issue by quinthar** [Expensify/App#52979]
**Background:** When a new user signs up, Concierge assigns tasks to the user based on their onboarding intentions.  Many of these tasks contain links.   We see in Fullstory that many users who are attempting to click the link, accidentally click next to it -- and thus are shown an "editor" for the description. 

**Problem:** When users accidentally edit the description of a task assigned to them, it unexpected and feels broken, undermining the perception of product quality and conversion.

**Solution:** Make all tasks read-only for everyone except the author.   This involves:
1. Prevent non-author editing on the front end
2. When done, we will update the API to prevent non-author editing on the back end.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Make all task titles/descriptions read-only to all but the author

### What is the root cause of that problem?
Improvment

### What changes do you think we should make in order to solve the problem?
Update `canModifyTask` function to return false if the `sessionAccountID` isn't the author of the task.

https://github.com/Expensify/App/blob/ef0d341c7d15ce1b636cf5bb70b1f389293af297/src/libs/actions/Task.ts#L1168

### What alternative solutions did you explore? (Optional)

**Comment by suneox** [comment]
> 1. Prevent non-author editing on the front end

@quinthar I’d like to confirm if this behavior only prevents editing tasks by Concierge while still allowing users to change the task status to complete/incomplete, or does it also prevent changing the task status?

**Comment by quinthar** [comment]
correct, the assignee CAN mark it as done, this only constrains editing the title & description.

**Comment by suneox** [comment]
> correct, the assignee CAN mark it as done, this only constrains editing the title & description.

Based on [this confirmation](https://github.com/Expensify/App/issues/52979#issuecomment-2509473131), we only prevent editing the title and description of tasks where the owner is Concierge. Both solutions from @nkdengineer and @Krishna2323 are do not meet the expected behavior and outdated with the latest main. We are still waiting for updates from the contributors.

> [!NOTE]  
> We only need solutions that match the confirmed behavior. Any solutions that do not align with [this behavior](https://github.com/Expensify/App/issues/52979#issuecomment-2509473131) should be removed to avoid cluttering the proposal with relevant options.

**Comment by suneox** [comment]
> @suneox, I added a [new proposal](https://github.com/Expensify/App/issues/52979#issuecomment-2510171819) and hid the previous one due to multiple edits.

@Krishna2323  Thank you for the update. Your solution looks good overall. However, we cannot update the task status directly from the Concierge chat.

https://github.com/user-attachments/assets/b56cab9c-d622-4b66-a3a1-39fa753cde76

Nice to have if you please share a test branch with new updates

**Comment by suneox** [comment]
The [solution](https://github.com/Expensify/App/issues/52979#issuecomment-2510171819) to add a new parameter to the `canModifyTask` function to check for ConciergeTask and restrict editing tasks LGTM. The arrangement of logic inside `canModifyTask` function can be handled during the PR review process. We can proceed with @Krishna2323 [proposal](https://github.com/Expensify/App/issues/52979#issuecomment-2510171819)

🎀 👀 🎀 C+ reviewed

**Comment by thienlnam** [comment]
Hmm, this was actually originally the way tasks were created but then we let the assignee update the task details and then we just let everyone who has the ability to comment on the task to edit it.

Since this problem is specific to Concierge, should we just restrict editing those tasks so that in other usages the assignee can still edit tasks? cc @quinthar

**Comment by quinthar** [comment]
In the long run we will likely make this configurable somehow.  But for now, we're thinking the easiest is to just change the default for all tasks.

**Comment by thienlnam** [comment]
Cool, sounds good

**Comment by suneox** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [x] 1z. Other: New feature request.

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [x] 2z. Other: New feature request.

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [x] 3z. Other: New feature request.

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: *N/A: This is a new feature request.*

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: *N/A*

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

    *N/A: Due to it isn't an impactful bug*

**Comment by mallenexpensify** [comment]
@suneox @Krishna2323 , can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~021875244713886279566

**Comment by suneox** [comment]
> @suneox @Krishna2323 , can you please accept the job and reply here once you have? 

@mallenexpensify  Thank you, I've accepted the offer

**Comment by mallenexpensify** [comment]
Contributor: @Krishna2323 paid $250 via Upwork
Contributor+: @suneox paid $250 via Upwork 

For the regression test, I'm unsure if one is needed so I'm going with @suneox 's recommendation above. 
> N/A: Due to it isn't an impactful bug 

If anyone disagrees, please comment and I can create one.  Thx
